geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/api/QueryPlanner.scala

```diff
@@ -240,6 +240,35 @@ object QueryPlanner extends LazyLogging {
     }
   }
 
+  def queryToTransformSFT(sft: SimpleFeatureType, properties: Seq[String]): SimpleFeatureType = {
```

Comment:
to avoid code duplication, this should return `(SimpleFeatureType, String)` and `setQueryTransform` should call this